src/main/java/io/github/zerthick/protectionperms/ProtectionPerms.java

```diff
@@ -78,13 +78,18 @@ public void onServerInit(GameInitializationEvent event) {
                 logger.warn("Error loading default config! Error: " + e.getMessage());
             }
         }
-
         //Load invert
         try {
-            CommentedConfigurationNode linksNode = configLoader.load().getNode("invert");
-            boolean invert = linksNode.getBoolean(false);
-            PermHandler.getInstance().init(invert);
-
+            CommentedConfigurationNode modeNode = configLoader.load().getNode("mode");
+            if (modeNode.isVirtual()) {
+                logger.warn("Config corrupt! Using defaults (whitelist mode)\nDelete " + instance.getId() + ".conf to fix");
+            }
+            String mode = modeNode.getString("whitelist");
```

Comment:
If the modeNode is virtual we should never try to fetch the value from it in the first place

Reply:
I could do
```java
if (modeNode.isVirtual()) {
    logger.warn("Config corrupt! Using defaults (whitelist mode)\nDelete " + instance.getId() + ".conf to fix");
    String mode = "whitelist";
} else {
    String mode = modeNode.getString("whitelist");
}
```

But since `getString()` allows a default value, I use the submitted method because it's simpler. (From a Pythonistic view)

Reply:
You're right